docs/Developers/Git_Basics.rst

```diff
@@ -70,7 +70,10 @@ Creating a new branch
 ---------------------
 
 To start working on an issue, you first need to create a new branch where you
-will work.
+will work. You should always have a clean and working branch which
+is considered to be the
+"master" branch. Also, developing on the master branch
+can potentially cause issues.
```

Comment:
Everything can be summed up in this IMO: 
`The default branch is "master" but when you want to add a new feature or fix a bug, no matter how big or how small, you spawn a new branch to encapsulate your changes. This makes sure that unstable code is never committed to the main code base, you have something stable at all times, and it gives you the chance to clean up your feature’s history before merging it into the main "master" branch`